x/bank/keeper/send.go

```diff
@@ -167,19 +167,37 @@ func (k BaseSendKeeper) InputOutputCoins(ctx context.Context, inputs []types.Inp
 		return err
 	}
 
-	sdkCtx := sdk.UnwrapSDKContext(ctx)
-
 	for _, in := range inputs {
 		inAddress, err := k.ak.AddressCodec().StringToBytes(in.Address)
 		if err != nil {
 			return err
 		}
 
 		inAddress = k.ak.GetMergedAccountAddressIfExists(ctx, inAddress)
+		for _, out := range outputs {
+			outAddress, err := k.ak.AddressCodec().StringToBytes(out.Address)
+			if err != nil {
+				return err
+			}
+
+			outAddress = k.ak.GetMergedAccountAddressIfExists(ctx, outAddress)
+			_, err = k.sendRestriction.apply(ctx, inAddress, outAddress, out.Coins)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	for _, in := range inputs {
+		inAddress, err := k.ak.AddressCodec().StringToBytes(in.Address)
```

Comment:
This can just check for inAddress for now but in the future when we change our SendRestriction to be slightly more complex we will need this nested loop